src/uct/ib/rc/base/rc_ep.c

```diff
@@ -459,23 +457,13 @@ ucs_status_t uct_rc_ep_flush(uct_rc_ep_t *ep, int16_t max_available,
 
 ucs_status_t uct_rc_ep_check_cqe(uct_rc_iface_t *iface, uct_rc_ep_t *ep)
 {
-    uct_rc_txqp_t *txqp;
-
     if (!uct_rc_iface_have_tx_cqe_avail(iface)) {
         UCS_STATS_UPDATE_COUNTER(iface->stats, UCT_RC_IFACE_STAT_NO_CQE, 1);
         UCS_STATS_UPDATE_COUNTER(ep->super.stats, UCT_EP_STAT_NO_RES, 1);
         return UCS_ERR_NO_RESOURCE;
     }
 
-    txqp = &ep->txqp;
-    /* if unsignaled == RC_UNSIGNALED_INF this value was already saved and \
-       next operation will be defenitly signaled */
-    if (txqp->unsignaled != RC_UNSIGNALED_INF) {
-        txqp->unsignaled_store_count++;
-        txqp->unsignaled_store += txqp->unsignaled;
-        txqp->unsignaled        = RC_UNSIGNALED_INF;
-    }
-
+    ep->txqp.unsignaled = RC_UNSIGNALED_INF;
```

Comment:
seems like it disables TX moderation...
BTW, is `RC_UNSIGNALED_INF` still needed?
I guess `ep->txqp.unsignaled` can contain always true value now (or 0 on failure since all next CQEs will be reported as "signaled") and we can count everything by pi/ci

Reply:
The objective was to retain both TX moderation and forced signaling in case of low free CQE count.
See no change in behavior. What do I miss?

Reply:
you are right, I missed that this value affects check in a caller macro
```
    if (ucs_unlikely((_iface)->tx.cq_available <= \
        (signed)(_iface)->config.tx_moderation)) { \
```
I would suggest to move this assignment and check to the same func/macro to make this logic more transparent if this is the only place which is affected by `RC_UNSIGNALED_INF ` now.